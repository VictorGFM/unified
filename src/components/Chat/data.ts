import { Message } from '../Message/Message'

export const messages: Message[] = [
  { text: 'Some user message...', author: 'Victor Moraes' },
  { text: 'Some contact message...', author: 'Contact Name' },
]
