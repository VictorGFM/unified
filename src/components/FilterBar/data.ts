import { FiltersGroupsData } from './FilterBar'

export const Filters: FiltersGroupsData = {
  Posts: [
    {
      id: '1',
      label: 'Instagram Posts',
    },
    {
      id: '2',
      label: 'Twitter Posts',
    },
    {
      id: '3',
      label: 'Facebook Posts',
    },
  ],
  Groups: [
    {
      id: '4',
      label: 'Grupo dos Amigos',
    },
    {
      id: '5',
      label: 'Grupo do Trabalho',
    },
    {
      id: '6',
      label: 'Grupo da Família',
    },
    {
      id: '7',
      label: 'Grupo dos Canários',
    },
  ],
  Others: [
    {
      id: '8',
      label: 'Other Filter',
    },
    {
      id: '9',
      label: 'Other Filter',
    },
    {
      id: '10',
      label: 'Other Filter',
    },
  ],
}
