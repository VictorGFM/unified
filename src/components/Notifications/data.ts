import { AlfredoPhotoIcon, ContactPhotoIcon } from '../../assets'
import { Notification } from '../NotificationCard/NotificationCard'

export const notifications: Notification[] = [
  {
    id: '1',
    contactName: 'Alfredo',
    contactPhoto: AlfredoPhotoIcon,
    description: 'liked your tweet.',
    time: '1h',
    socialType: 'Twitter',
  },
  {
    id: '2',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '1h',
    socialType: 'Instagram',
  },
  {
    id: '3',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'shared your post.',
    time: '1h',
    socialType: 'Facebook',
  },
  {
    id: '4',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '1h',
    socialType: 'Instagram',
  },
  {
    id: '5',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '2h',
    socialType: 'Twitter',
  },
]
