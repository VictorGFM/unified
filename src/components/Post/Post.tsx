import { CommentIcon, LikeIcon, ProfilePhotoIcon, ShareIcon } from '../../assets'
import PostButton from '../PostButton'
import SocialTag from '../SocialTag'
import {
  ButtonsDiv,
  ContactName,
  ContactPhoto,
  ContactPhotoImage,
  PostDescription,
  PostDiv,
  PostImage,
  PostTime,
  PostHeader,
  PostInfoDiv,
  ContactNameDiv,
} from './styles'

export interface NumberInteractions {
  likes: number
  comments: number
  shares: number
}

export interface PostInterface {
  id: string
  contactName: string
  contactPhoto: string
  contactGroup?: string
  postTime: string
  socialType: string
  postDescription: string
  postImage?: string
  numberInteractions: NumberInteractions
}

interface PostProps {
  post: PostInterface
}

const Post = ({ post }: PostProps) => {
  const {
    contactName,
    contactPhoto,
    postTime,
    socialType,
    postDescription,
    postImage,
    numberInteractions,
  } = post
  return (
    <PostDiv>
      <PostHeader>
        <ContactPhoto>
          <ContactPhotoImage src={contactPhoto} />
        </ContactPhoto>
        <PostInfoDiv>
          <ContactNameDiv>
            <ContactName>{contactName}</ContactName>
            <PostTime>{postTime}</PostTime>
          </ContactNameDiv>
          <PostDescription>{postDescription}</PostDescription>
          <SocialTag socialType={socialType} />
        </PostInfoDiv>
      </PostHeader>
      {postImage && <PostImage src={postImage} />}
      <ButtonsDiv>
        <PostButton buttonImage={LikeIcon} numberInteractions={numberInteractions.likes} />
        <PostButton buttonImage={CommentIcon} numberInteractions={numberInteractions.comments} />
        <PostButton buttonImage={ShareIcon} numberInteractions={numberInteractions.shares} />
      </ButtonsDiv>
    </PostDiv>
  )
}

export default Post
