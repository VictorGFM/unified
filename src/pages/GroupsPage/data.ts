import { GroupImage1, GroupImage2, GroupImage3, GroupImage4 } from '../../assets'
import { Group } from '../../components/GroupCard/GroupCard'

export const groups: Group[] = [
  { id: '1', title: 'Grupo dos Amigos', image: GroupImage1 },
  { id: '2', title: 'Grupo da Família', image: GroupImage2 },
  { id: '3', title: 'Grupo do Trabalho', image: GroupImage3 },
  { id: '4', title: 'Grupo dos Canários', image: GroupImage4 },
]
