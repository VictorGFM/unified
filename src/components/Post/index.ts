import Post from './StoriesBar'

export default Post
