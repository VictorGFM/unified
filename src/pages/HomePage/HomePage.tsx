import FilterBar from '../../components/FilterBar'
import StoriesBar from '../../components/StoriesBar'
import Post from '../../components/Post'
import { HomePageDiv } from './styles'
import { posts } from './data'

const HomePage = () => {
  return (
    <HomePageDiv>
      <FilterBar />
      <StoriesBar />
      {posts?.map(post => (
        <Post key={post.id} post={post} />
      ))}
    </HomePageDiv>
  )
}

export default HomePage
