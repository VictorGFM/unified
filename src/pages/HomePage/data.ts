import { NumberInteractions, PostInterface } from '../../components/Post/Post'
import { AlfredoPhotoIcon, AndromedaImage, ContactPhotoIcon, ProfilePhotoIcon } from '../../assets'

export const interactions: NumberInteractions[] = [
  {
    likes: 237,
    comments: 15,
    shares: 10,
  },
  {
    likes: 107,
    comments: 10,
    shares: 7,
  },
  {
    likes: 5,
    comments: 2,
    shares: 1,
  },
  {
    likes: 527,
    comments: 132,
    shares: 97,
  },
]

export const posts: PostInterface[] = [
  {
    id: '1',
    contactName: 'Victor Moraes',
    contactPhoto: ProfilePhotoIcon,
    contactGroup: 'Grupo da Família',
    postTime: '1h',
    socialType: 'Instagram',
    postDescription: 'Look at this galaxy! Its so beautiful...',
    postImage: AndromedaImage,
    numberInteractions: interactions[0],
  },
  {
    id: '2',
    contactName: 'Victor Moraes',
    contactPhoto: ProfilePhotoIcon,
    contactGroup: 'Grupo da Família',
    postTime: '3h',
    socialType: 'Twitter',
    postDescription:
      'So, this is a description of a Twitter post, but right after an Instagram post. All in the same place, awesome,' +
      ' isnt it?',
    numberInteractions: interactions[1],
  },
  {
    id: '3',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    contactGroup: 'Grupo dos Canários',
    postTime: '15h',
    socialType: 'Facebook',
    postDescription: 'Some post description...',
    numberInteractions: interactions[2],
  },
  {
    id: '4',
    contactName: 'Alfredo',
    contactPhoto: AlfredoPhotoIcon,
    contactGroup: 'Grupo dos Amigos',
    postTime: '21h',
    socialType: 'Twitter',
    postDescription: 'Au Au Au...',
    numberInteractions: interactions[3],
  },
]
