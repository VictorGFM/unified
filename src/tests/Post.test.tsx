import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import Post from '../components/Post'
import { posts } from '../pages/HomePage/data'

describe('<Post/>', () => {
  it('should show match snapshot', () => {
    const component = render(<Post post={posts[0]} />)

    expect(component).toMatchSnapshot()
  })
})
