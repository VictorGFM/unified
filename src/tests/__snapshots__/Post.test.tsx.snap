// Vitest Snapshot v1

exports[`<Post/> > should show match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gKXOVf gsDAIt"
      >
        <div
          class="sc-iBkjds fyqkwp"
        >
          <div
            class="sc-ftvSup iOGfLH"
          >
            <img
              class="sc-papXJ hbQSzU"
              src="/src/assets/icons/ProfilePhotoIcon.svg"
            />
          </div>
          <div
            class="sc-jqUVSM iJoCqh"
          >
            <div
              class="sc-kDDrLX TEqDf"
            >
              <span
                class="sc-iqcoie hzGzAq"
              >
                Victor Moraes
              </span>
              <span
                class="sc-crXcEl gjdcae"
              >
                1h
              </span>
            </div>
            <span
              class="sc-evZas hHCdeT"
            >
              Look at this galaxy! Its so beautiful...
            </span>
            <div
              class="sc-eCYdqJ jtafVE"
            >
              <img
                class="sc-jSMfEi jwEDwO"
                src="/src/assets/icons/InstagramIcon.svg"
              />
            </div>
          </div>
        </div>
        <img
          class="sc-breuTD jbscYM"
          src="/src/assets/images/AndromedaImage.png"
        />
        <div
          class="sc-ksZaOG izmBSs"
        >
          <div
            class="sc-bczRLJ kfAyjk"
          >
            <div
              class="sc-gsnTZi fHNkQO"
            >
              <img
                class="sc-dkzDqf KaWxl"
                src="/src/assets/icons/LikeIcon.svg"
              />
            </div>
            <span
              class="sc-hKMtZM esTaXM"
            >
              237
            </span>
          </div>
          <div
            class="sc-bczRLJ kfAyjk"
          >
            <div
              class="sc-gsnTZi fHNkQO"
            >
              <img
                class="sc-dkzDqf KaWxl"
                src="/src/assets/icons/CommentIcon.svg"
              />
            </div>
            <span
              class="sc-hKMtZM esTaXM"
            >
              15
            </span>
          </div>
          <div
            class="sc-bczRLJ kfAyjk"
          >
            <div
              class="sc-gsnTZi fHNkQO"
            >
              <img
                class="sc-dkzDqf KaWxl"
                src="/src/assets/icons/ShareIcon.svg"
              />
            </div>
            <span
              class="sc-hKMtZM esTaXM"
            >
              10
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gKXOVf gsDAIt"
    >
      <div
        class="sc-iBkjds fyqkwp"
      >
        <div
          class="sc-ftvSup iOGfLH"
        >
          <img
            class="sc-papXJ hbQSzU"
            src="/src/assets/icons/ProfilePhotoIcon.svg"
          />
        </div>
        <div
          class="sc-jqUVSM iJoCqh"
        >
          <div
            class="sc-kDDrLX TEqDf"
          >
            <span
              class="sc-iqcoie hzGzAq"
            >
              Victor Moraes
            </span>
            <span
              class="sc-crXcEl gjdcae"
            >
              1h
            </span>
          </div>
          <span
            class="sc-evZas hHCdeT"
          >
            Look at this galaxy! Its so beautiful...
          </span>
          <div
            class="sc-eCYdqJ jtafVE"
          >
            <img
              class="sc-jSMfEi jwEDwO"
              src="/src/assets/icons/InstagramIcon.svg"
            />
          </div>
        </div>
      </div>
      <img
        class="sc-breuTD jbscYM"
        src="/src/assets/images/AndromedaImage.png"
      />
      <div
        class="sc-ksZaOG izmBSs"
      >
        <div
          class="sc-bczRLJ kfAyjk"
        >
          <div
            class="sc-gsnTZi fHNkQO"
          >
            <img
              class="sc-dkzDqf KaWxl"
              src="/src/assets/icons/LikeIcon.svg"
            />
          </div>
          <span
            class="sc-hKMtZM esTaXM"
          >
            237
          </span>
        </div>
        <div
          class="sc-bczRLJ kfAyjk"
        >
          <div
            class="sc-gsnTZi fHNkQO"
          >
            <img
              class="sc-dkzDqf KaWxl"
              src="/src/assets/icons/CommentIcon.svg"
            />
          </div>
          <span
            class="sc-hKMtZM esTaXM"
          >
            15
          </span>
        </div>
        <div
          class="sc-bczRLJ kfAyjk"
        >
          <div
            class="sc-gsnTZi fHNkQO"
          >
            <img
              class="sc-dkzDqf KaWxl"
              src="/src/assets/icons/ShareIcon.svg"
            />
          </div>
          <span
            class="sc-hKMtZM esTaXM"
          >
            10
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
